lib/elixir/lib/gen_server.ex

```diff
@@ -271,6 +271,14 @@ defmodule GenServer do
   generated atoms won't be garbage-collected. For such cases, you can
   set up your own local registry by using the `Registry` module.
 
+  For example:
+
+      {:ok, _} = Registry.start_link(keys: :unique, name: :stacks)
+      name = {:via, Registry, {:stacks, "stack 1"}}
+      {:ok, _pid} = GenServer.start_link(Stack, "hello", name: name)
+      Registry.lookup(:stacks, "stack 1")
+      #=> [{#PID<0.150.0>, nil}]
```

Comment:
@nathanl `Registry.lookup/2` is a dangerous pitfall (ask me how I know...) because it won't work if you attach metadata to the `:via` tuple. Maybe we use `GenServer.whereis/1` in this example?

Reply:
@whatyouhide I'm trying to add an example of the advice in the previous paragraph regarding `Registry`. `GenServer.whereis/1` is for global atom names only, right?

I wasn't aware that `Registry` could store metadata, but it looks like `Registry.lookup/2` can fetch it.

```elixir
metadata = "taters"
{:ok, _} = Registry.register(:stacks, 9, metadata)
Registry.lookup(:stacks, 9)
[{#PID<0.138.0>, "taters"}] # value == "metadata"?
 Registry.lookup(:stacks, :bob) # registered with a :via tuple
[{#PID<0.137.0>, nil}]
```

I see in the v1.19.0-dev docs that `{:via` will be able to include metadata soon - is that when `lookup/2` becomes a problem? What am I missing?

Reply:
Yes that's when `lookup/2` becomes a problem. I would just register via `:via` here but use `GenServer.whereis` instead of `Registry.lookup`.

Reply:
```suggestion
      GenServer.whereis(name)
      #=> #PID<0.150.0>
```